lib/Fetcher/FeedFetcher.php

```diff
@@ -256,6 +256,14 @@ protected function buildItem(ItemInterface $parsedItem, ?string $body = null, bo
             $item->setAuthor($this->decodeTwice($author->getName()));
         }
 
+        $categories = [];
+        foreach ($parsedItem->getCategories() as $category) {
+            $categories[] = $this->decodeTwice($category->getLabel());
+        }
+        if (count($categories) > 0) {
+            $item->setTags(implode(',', $categories));
```

Comment:
You can do an implode on a single item without any affect AFAIK. So this check isn't needed

Reply:
It's checking for an empty array here, since `implode` would then result in an empty string which we don't want in the database.

Maybe it could also be checked using `!== []` or the logic could be moved into the `Item` class so `setTags` would take an array and `getTags` could return an string-array instead, since this would be the main use-case.
Otherwise every frontend or API implementation would need to `explode` on its own.

Reply:
That sounds good to me although you might just be able to let doctrine handle all that by making it an array?

Reply:
Did some research on both methods, and I'm not sure that nextcloud supports storing data as arrays.. I've tried implementing a [doctrine simple_array](https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/types.html#simple-array), but I get weird behavoir, such as a string with the value "Array" being inserted.. It then also fails in the Entity->fromRow() function..
I've dug deep and to my understanding, [the array type isn't even supported](https://github.com/nextcloud/server/blob/e559afb8d409f75fdf9a216428d858d08aa1ee03/lib/public/AppFramework/Db/QBMapper.php#L226-L247)... I don't know if it's just me, I'll try to look further into it, I'm posting here just in case someone already had experience with this, I'd appreciate any suggestions, thanks

Reply:
I've also done some research and I haven't found an easy way to write an array directly to a string DB field using the Doctrine methods here in NC.

My approach would still be something like that, but I'm not sure how @SMillerDev is thinking about that:
<details><summary>Simplified setter and getter</summary>

```php
    /** @var string|null */
    protected $categories;

    public function getCategories(): ?array
    {
        if ($this->categories === null)
            return null;

        return explode(',', $this->categories);
    }

    public function setCategories(array $categories = null): self
    {
        $serializedCategories = ($categories !== null && $categories !== []) ? implode(',', $categories) : null;
        if ($this->categories !== $serializedCategories) {
            $this->categories = $serializedCategories;
            $this->markFieldUpdated('categories');
        }

        return $this;
    }
```
(Alternatively we could also use `json_encode`/`json_decode` to serialize the array, which would then also allow `,` inside the category name)
</details>

Reply:
I like serializing to json. Maybe doctrine has a json field type (because a lot of databases do) so it can still be a stricter representation of the data than just string?

Reply:
I've tried implementing it your way as well, @anoymouserver, however I still get issues, mainly because the setter now expects an array as an argument, and when loading the items from the database a function called Entity::fromRow() is called, which calls the setter and passes the value found in the db (string) as an argument..
But I think I figured it out, in addition to using JSON..

We'll add a field called categoriesJSON which is stored as text in the DB, and those getters and setters (`getCategoriesJSON`, `setCategoriesJSON`) will use strings, so there will be no issues when loading them.. In addition to that we'll add two more getters and setters `setCategories` and `getCategories` which return an array and take an array as an argument respectively, and will use the JSON setters.

This method was inspired by the way the [forms app deals with JSON fields](https://github.com/nextcloud/forms/blob/master/lib/Db/Form.php)

Reply:
Oh, interesting, I didn't know that they are implicitly called by the Entity base class.